File: app/src/main/java/org/stepic/droid/core/presenters/contracts/CourseJoinView.java
package org.stepic.droid.core.presenters.contracts;

import org.stepic.droid.events.joining_course.FailJoinEvent;
import org.stepic.droid.events.joining_course.SuccessJoinEvent;

public interface CourseJoinView {
    void showProgress();

    void setEnabledJoinButton(boolean isEnabled);

    void onFailJoin(FailJoinEvent e);

    void onSuccessJoin(SuccessJoinEvent e);
}

File: app/src/main/java/org/stepic/droid/core/presenters/contracts/CourseJoinView.kt
package org.stepic.droid.core.presenters.contracts

import org.stepik.android.model.Course

interface CourseJoinView {
    fun showProgress()

    fun setEnabledJoinButton(isEnabled: Boolean)

    fun onFailJoin(code: Int)

    fun onSuccessJoin(course: Course)
}
